docs/_static/nuts-discovery.yaml

```diff
@@ -0,0 +1,115 @@
+openapi: "3.0.0"
+info:
+  title: Nuts discovery API spec
+  description: API specification for certificate related services hosted by the network authority
+  version: 0.1.0
+  license:
+    name: GPLv3
+paths:
+  /api/csr:
+    post:
+      operationId: submit
+      summary: "submit a PEM encoded CSR to the network authority"
+      tags:
+        - certificates
+      requestBody:
+        description: PEM encoded CSR
+        required: true
+        content:
+          text/plain:
+            schema:
+              type: string
+      responses:
+        '200':
+          description: "request was accepted"
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/certificate_request"
+  /api/csr/{urn}:
+    get:
+      operationId: listRequests
+      summary: "list certificate signing requests"
+      tags:
+        - certificates
+      parameters:
+        - name: urn
+          in: path
+          description: "vendor oid in urn format"
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: "success"
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/certificate_request"
+  /api/x509/{urn}:
+    get:
+      operationId: listCertificates
+      summary: "get signed certificate(s) based on urn. The urn must be an oid which corresponds to the oid in the SAN"
+      tags:
+        - certificates
+      parameters:
+        - name: urn
+          in: path
+          description: "vendor oid in urn format"
+          required: true
+          schema:
+            type: string
+      responses:
+        200:
+          description: "List of signed certificates for given urn. The certificates might be for different keys and/or different validity. The client should match them."
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/certificateWithChain"
+        404:
+          description: "No certificates found for given urn"
+          content:
+            text/plain:
+              schema:
+                type: string
+components:
+  schemas:
+    certificate_request:
+      required:
+        - name
+        - oid
+        - pem
+        - submittedAt
+      properties:
+        name:
+          type: string
+          description: "CN of request"
+        oid:
+          type: string
+          description: "oid of vendor, can be used to query requests"
+        pem:
+          type: string
+          description: "the pem-encoded CSR"
+        submittedAt:
```

Comment:
isn't there a date/time format/type in OpenAPI?

Reply:
code generation explodes cross language